### DISCUSSION TITLE
HELP,when i use "etcdctl member list" i got error

### DISCUSSION THREAD

**Discussion by Liby798** [etcd-io/etcd#15228] [Q&A]
`{"level":"warn","ts":"2023-01-31T18:51:15.491+0800","logger":"etcd-client","caller":"v3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc00034aa80/127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:2379: connect: connection refused\""}
Error: context deadline exceeded`
This error was reported to me when I used this command, I don't know how to solve it, please help me

**Comment by Liby798** [comment]
I intend to do cluster deployment, but three services are deployed now, and all three services are running. I would like to know whether the error in the cluster status report indicates that the cluster is not built successfully.
I checked the health status of a single node is normal
This is the command I use to check my health status
`etcdctl --endpoints ip:2379 --cacert=/etc/ssl/etcd/ssl/ca.crt --cert=/etc/ssl/etcd/ssl/etcd.crt --key=/etc/ssl/etcd/ssl/etcd.key endpoint health`

**Comment by SimFG** [comment]
From the error message, I guess that the connection is refused.
Try to add some params about tls, like:
`etcdctl --endpoints ip:2379 --cacert=/etc/ssl/etcd/ssl/ca.crt --cert=/etc/ssl/etcd/ssl/etcd.crt --key=/etc/ssl/etcd/ssl/etcd.key member list`

**Reply by Liby798** [reply]
thx,but I have one more question,when I use the command you suggested, the information returned by the three nodes is inconsistent. I would like to know the reason for this and whether it has any impact on my cluster construction.
This is what I found using the first node ip:
`etcdctl --endpoints ip1:2379 --cacert=/etc/ssl/etcd/ssl/ca.crt --cert=/etc/ssl/etcd/ssl/etcd.crt --key=/etc/ssl/etcd/ssl/etcd.key member list`
`d74273182fdee3e0, started, etcd1, https://ip1:2380, https://ip1:2379, false`
This is what I found using the second node ip:
`etcdctl --endpoints ip2:2379 --cacert=/etc/ssl/etcd/ssl/ca.crt --cert=/etc/ssl/etcd/ssl/etcd.crt --key=/etc/ssl/etcd/ssl/etcd.key member list`
`475d6d67f41f6f87, started, etcd3, https://ip3:2380, https://ip3:2379, false
55a484bcdbb509ad, started, etcd2, https://ip2:2380, https://ip2:2379, false
d74273182fdee3e0, started, etcd1, https://ip1:2380, , false`
The third ip node came back with the same result as the second.
It can be seen that the first ip check only returned its own condition, but the second and third ip check returned the condition of all nodes, and the return result of the first node's port 2379 is missing, I am not sure whether this is normal.
Please help me answer it if you are convenient. Thank you

**Comment by SimFG** [comment]
sorry, I used the endpoints param wrongly. You should like this:
`etcdctl --endpoints=${HOST_1}:2379,${HOST_2}:2379,${HOST_3}:2379 -cacert=/etc/ssl/etcd/ssl/ca.crt --cert=/etc/ssl/etcd/ssl/etcd.crt --key=/etc/ssl/etcd/ssl/etcd.key member list`
reference article: https://etcd.io/docs/v3.5/tutorials/how-to-deal-with-membership/

**Reply by Liby798** [reply]
well,I tried it as you said, but the return result is still missing port 2379 of ip1 node, is that right？
`etcdctl --endpoints=ip1:2379,ip2:2379,ip3:2379 --cacert=/etc/ssl/etcd/ssl/ca.crt --cert=/etc/ssl/etcd/ssl/etcd.crt --key=/etc/ssl/etcd/ssl/etcd.key member list`
`475d6d67f41f6f87, started, etcd3, https://ip3:2380, https://ip3:2379, false
55a484bcdbb509ad, started, etcd2, https://ip2:2380, https://ip2:2379, false
d74273182fdee3e0, started, etcd1, https://ip1:2380, , false`
and I tried it on three nodes, and it returned the same value

**Reply by SimFG** [reply]
I think you may miss some params when you launched this cluster. reference: https://etcd.io/docs/v3.5/tutorials/how-to-setup-cluster/

**Reply by Liby798** [reply]
maybe ,i'll check again and by the way, thx
